portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -13,42 +13,92 @@
-    facts = new ArrayList<>();
-    facts.add(
-        "My favorite color is green. "
-            + "I also like light blue.");
-    facts.add("My parents made up my name based on my Chinese one.");
-    facts.add("My birthday is on Halloween.");
-    facts.add("I have the best younger brother!");
-    facts.add("My favorite food is pasta.");
-    facts.add("I'm a cat person. I like dogs, but I like cats more.");
-    facts.add(
-        "I didn't have any experience with "
-            + "computer science in high school.");
-    facts.add("I play the acoustic guitar.");
+    comments = new ArrayList<>();
   }
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    String fact = facts.get((int) (Math.random() * facts.size()));
-    response.setContentType("text/html;");
-    response.getWriter().println(fact);
+    Query query = new Query("Task");
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    // Loop over entities.
+    List<Task> tasks = new ArrayList<>();
+    for (Entity entity : results.asIterable()) {
+      String name = (String) entity.getProperty("name");
+      String text = (String) entity.getProperty("text");
+      long timestamp = (long) entity.getProperty("timestamp");
+      tasks.add(new Task(name, text, timestamp));
+    }
+
+    response.setContentType("application/json;");
+    response.getWriter().println(gson.toJson(tasks));
+  }
+
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get the input from the form.
+    String name = getParameter(request, "name-input", "");
+    String text = getParameter(request, "text-input", "");
+    long timestamp = System.currentTimeMillis();
+
+    // Create new Entity with kind Task and set properties with keys and values.
+    Entity taskEntity = new Entity("Task");
+    taskEntity.setProperty("name", name);
+    taskEntity.setProperty("text", text);
+    taskEntity.setProperty("timestamp", timestamp);
+
+    // Create instance of DatastoreService class and store entity.
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    datastore.put(taskEntity);
+
+    // Respond with the result.
+    response.sendRedirect("/comments.html");
+  }
+    /**
+   * @return the request parameter, or the default value if the parameter
+   *         was not specified by the client.
+   */
+  private String getParameter(HttpServletRequest request, String name, String defaultValue) {
+    String value = request.getParameter(name);
+    if (value == null) {
+      return defaultValue;
+    }
+    return value;
+  }
+  /**
+  * Converts a ServerStats instance into a JSON string using the Gson library. Note: We first added
+  * the Gson library dependency to pom.xml.
+  */
+  private String convertToJsonUsingGson(ArrayList<String> comments) {
```

Comment:
we can actually just get rid of this method and just do GSON.toJson() directly where ever convertToJsonUsingGson() is currently used